src/yunohost/certificate.py

```diff
@@ -809,6 +816,15 @@ def _domain_is_accessible_through_HTTP(ip, domain):
     return True
 
 
+def _domain_is_resolved_locally(domain):
+    try:
+        requests.head("http://%s/" % domain)
```

Comment:
We should try to resolve the server using a DNS query and checking that this is exactly this problem, not by doing a HTTP request (this can trigger other problems that aren't related to DNS and the catchall exception can hide other errors).